pkg/remote/remote.go

```diff
@@ -0,0 +1,52 @@
+package remote
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/okteto/okteto/pkg/discovery"
+	"github.com/spf13/afero"
+)
+
+const (
+	oktetoDockerignoreName = ".oktetodeployignore"
+)
+
+func CreateDockerignoreFileWithFilesystem(cwd, tmpDir, manifestPathFlag string, fs afero.Fs) error {
+	dockerignoreContent := []byte(``)
+	dockerignoreFilePath := filepath.Join(cwd, oktetoDockerignoreName)
+	if _, err := fs.Stat(dockerignoreFilePath); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
```

Comment:
As we discussed this if needs an update, it's not going in this condition when passing a non existing `cwd`.

```suggestion
		if os.IsNotExist(err) {
```

I've found this one works, but please double check 🙏

Reply:
This is a test case I'd add:

```go
func TestCreateDockerignoreFileWithFilesystem_ErrNotExist(t *testing.T) {
	tempDir := t.TempDir()
	fs := afero.NewMemMapFs()
	err := CreateDockerignoreFileWithFilesystem("/non-existing-test-folder", tempDir, "okteto.yml", fs)
	assert.ErrorIs(t, err, os.ErrNotExist)
}
```

Reply:
I am not 100% sure of my suggestion because afero seems to return `fs.PathError` for not found, and inspecting the code of `os.IsNotExist` it checks against `fs.ErrNotExist`

Reply:
I'd also check the implementation of `FileExistsWithFilesystem` which seems doing what I was suggesting (using `os.IsNotExist`

Reply:
I've checked by debugging and the err is wrapping an non exist error. We want to return err only if the err is not a not exist one.

In order to add a test case i am not sure how i can mock afero to return other err ?

Reply:
@teresaromero if we pass a non existing path, shouldn't that be a not found error?

Reply:
It is a non-exist error. The file does not exist on that path, so we create an empty dockerignore for the build 

![Screenshot 2023-07-03 at 10 22 49](https://github.com/okteto/okteto/assets/40767058/2c2455a0-722b-4b09-9708-cccbf3c535af)